library/src/blas2/rocblas_syr.cpp

```diff
@@ -94,38 +102,33 @@ namespace
 
 extern "C" {
 
-rocblas_status rocblas_ssyr(rocblas_handle handle,
-                            rocblas_fill   uplo,
-                            rocblas_int    n,
-                            const float*   alpha,
-                            const float*   x,
-                            rocblas_int    incx,
-                            float*         A,
-                            rocblas_int    lda)
-try
-{
-    return rocblas_syr_impl(handle, uplo, n, alpha, x, incx, A, lda);
-}
-catch(...)
-{
-    return exception_to_rocblas_status();
-}
+#ifdef IMPL
+#error IMPL ALREADY DEFINED
+#endif
 
-rocblas_status rocblas_dsyr(rocblas_handle handle,
-                            rocblas_fill   uplo,
-                            rocblas_int    n,
-                            const double*  alpha,
-                            const double*  x,
-                            rocblas_int    incx,
-                            double*        A,
-                            rocblas_int    lda)
-try
-{
-    return rocblas_syr_impl(handle, uplo, n, alpha, x, incx, A, lda);
-}
-catch(...)
-{
-    return exception_to_rocblas_status();
-}
+#define IMPL(routine_name_, T_)                                           \
```

Comment:
I think templates would be better than macros.

Reply:
I will try to redo with a template.

Reply:
?

Reply:
Can't see how to easily replace the extern C symbols with an instatiated template.  This pattern exists elsewhere so a new PR can be made if Lee or someone else shows a good template pattern for these C APIs to be generated.  Resolving this conversation.   Use the rocblas project if making a ticket for this.

Reply:
> Can't see how to easily replace the extern C symbols with an instatiated template. This pattern exists elsewhere so a new PR can be made if Lee or someone else shows a good template pattern for these C APIs to be generated. Resolving this conversation. Use the rocblas project if making a ticket for this.

You can't create `extern "C"` template instantiations, since their names are always C++-binding (i.e., mangled), but you can avoid some of the repetitious code such as the `try/catch` block with a template.

I imagine this will come up in the solver proposal. I'll comment more later.